providers/closecrm/search.go

```diff
@@ -0,0 +1,99 @@
+package closecrm
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	"github.com/amp-labs/connectors/common"
+)
+
+/*
+Response Schema:
+
+{
+  "data": [{...},{...}]
+  cursor: "..."
+
+}
+
+*/
+
+// searchEndpoint represents the search api endpoint.
+// ref: https://developer.close.com/resources/advanced-filtering/
+const searchEndpoint = "data/search"
+
+// Search reads data through searching API. Supports advanced filtering using the filters field.
+// The NextPage Token generated takes 30 seconds to expire.
+//
+// doc: https://developer.close.com/resources/advanced-filtering
+func (c *Connector) Search(ctx context.Context, config SearchParams) (*common.ReadResult, error) {
+	input, err := buildUpdatedDateFilter(config)
+	if err != nil {
+		return nil, err
+	}
+
+	url, err := c.getAPIURL(searchEndpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := c.Client.Post(ctx, url.String(), input)
+	if err != nil {
+		return nil, err
+	}
+
+	return common.ParseResult(
+		resp,
+		common.GetRecordsUnderJSONPath("data"),
+		searchNextRecords(),
```

Comment:
unnecessary function invocation to get an anonymous function. See the comment near `searchNextRecords` definition and fix this.